### DISCUSSION TITLE
Switch between table and kanban views

### DISCUSSION THREAD

**Discussion by binaryfire** [mokhosh/filament-kanban#42] [Ideas]
Another suggestion for v2 would be a view switch in the top right of the header (similar in Adam's Curator plugin) which allows switching between the kanban view and a regular, fully customizable table view.

Pretty much every to-do app on the market has this and it's really useful. For example:

ClickUp board view: https://clickup.com/features/kanban-board
ClickUp table view: https://clickup.com/features/table-view

This plugin might provide some inspiration (it was extracted from Curator as a stand-alone thing): https://filamentphp.com/plugins/tgeorgel-table-layout-toggle

**Comment by mokhosh** [comment]
Thanks for the idea. I'll try to see if we can make it happen.

**Reply by sasaorasanin** [reply]
Hey @mokhosh, I was looking into this a bit and I found out some way to use all table functionality with custom view by setting content. Here is a simple code example how it works.

```php

public static function table(Table $table): Table
    {
        return $table
            ->content(fn ($livewire) => $livewire->getTableFilterState('viewType')['value'] == 'kanban' ? view('kanban') : null)
            ->filters([
                Tables\Filters\SelectFilter::make('viewType')
                    ->query(fn ($query) => $query) // if $state is kanban group by status/column for kanban if needed
                    ->label('View Type')
                    ->options([
                        'table' => 'List',
                        'kanban' => 'Kanban',
                    ])
                    ->default('table'),
            ])
            ->filtersLayout(FiltersLayout::AboveContent)
            ->columns([
                Tables\Columns\TextColumn::make('name')
                    ->label('Name')
                    ->searchable()
                    ->sortable(),
                Tables\Columns\TextColumn::make('parent.name')
                    ->label('Parent')
                    ->searchable()
                    ->sortable(),
                Tables\Columns\IconColumn::make('is_visible')
                    ->label('Visibility')
                    ->sortable(),
                Tables\Columns\TextColumn::make('updated_at')
                    ->label('Updated Date')
                    ->date()
                    ->sortable(),
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
            ])
            ->groupedBulkActions([
                Tables\Actions\DeleteBulkAction::make()
                    ->action(function () {
                        Notification::make()
                            ->title('Now, now, don\'t be cheeky, leave some records for others to play with!')
                            ->warning()
                            ->send();
                    }),
            ]);
    }
```

```php
// kanban.blade.php
@foreach ($records as $record)
    <p>{{ $record->name }}</p>
@endforeach
```
![image](https://github.com/user-attachments/assets/5f5e93ac-c365-4f2c-b75b-91d2cf1d5818)


In kanban we could pass your kanban component.

I mean this way it would be as part of the table which can be good and bad. But it would have all table abilities from filters etc..

## There is some random checkbox in filters for selecting all rows but since those would  not be rows, that can be hidden  i guess.

```php
->selectable(fn ($livewire) => $livewire->getTableFilterState('viewType')['value'] !== 'kanban')
```

**Reply by mokhosh** [reply]
Thanks @InfinityXTech for the exploration.
Do you mind creating a demo repo for this so I can check it out?

**Reply by sasaorasanin** [reply]
I almost made it to work as livewire component directly instead of page.
![image](https://github.com/user-attachments/assets/54241ed4-9892-45b0-85ed-41cd01775e24)

But there are few things thay may need to be checked like why doesnt livewire component doesn reload when pagination changes, could be because of `wire.ignore`. Or is pagination needed at all since it is kanban, it wouldnt paginate same amount of records per status unless there is custom logic.

I will share simpler repo with these changes later today. I guess this would be part of v3, when will v3 come out? :D

**Reply by sasaorasanin** [reply]
@mokhosh here is repo: https://github.com/InfinityXTech/filament-table-kanban

**Reply by mokhosh** [reply]
@InfinityXTech Thanks for creating the repo.

The Kanban view is not functional right? It just shows a static view. Or am I missing something?

**Reply by sasaorasanin** [reply]
It is not fully functional, just changing statuses. Modal doesnt. I pushed some things that I changed before to make it work. For example: 

`<div x-data wire:ignore class="md:flex overflow-x-auto overflow-y-hidden gap-4 pb-4">`

it was `wire:ignore.self`

if I leave it with self, when you move between statuses, it moves them back to old ones until you refresh the page. When everything is ignored it works, but some other things may not work.

**Reply by mokhosh** [reply]
Hmm... Seems like needs a deeper digging.
